packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -218,21 +216,33 @@ export class SummaryManager implements IDisposable {
             const shouldSummarizeState = this.getShouldSummarizeState();
             if (shouldSummarizeState.shouldSummarize === false) {
                 this.state = SummaryManagerState.Starting;
-                summarizer.stop(shouldSummarizeState.stopReason);
-                return;
+                summarizer.stop(shouldSummarizeState.stopReason, true /* forceClose */);
+                return "early exit 2";
```

Comment:
"force close after starting summarizer" or something like that may be more descriptive here.